buildSrc/src/main/java/org/elasticsearch/gradle/testclusters/ElasticsearchNode.java

```diff
@@ -833,6 +834,56 @@ private void checkFrozen() {
         }
     }
 
+    private Path getExtractedDistributionDir() {
+        return artifactsExtractDir.resolve(distribution.getGroup()).resolve("elasticsearch-" + getVersion());
+    }
+
+    @Classpath
+    private Set<File> getRuntimeClasspath() {
+        return project.fileTree(getExtractedDistributionDir()).matching(filter -> filter.include("**/*.jar")).getFiles();
+    }
+
+    @InputFiles
+    @PathSensitive(PathSensitivity.RELATIVE)
+    private FileCollection getDistributionFiles() {
```

Comment:
This will run before we start clusters, so won't have any of the files we create as part of startup ( like the config file ). That's why we need to track the extra config files. 
This all works as intended, just wanted to leave it here for other reviewers.

Reply:
I think this actually opens up another issue, which is the installed plugins and modules won't be there either. I _thought_ I tested this, but I need to verify.

Reply:
Yep, I totally missed the fact that these get installed _after_ the node starts. So my test build scan above "works" because the plugin under test is also on the test classpath but this won't work in all cases as we may install plugins/modules that aren't on the test runtime classpath. I'm lookin into a solution for this.

Reply:
I've pushed a solution here. Essentially, we grap the bundle archives for each plugin/module and snapshot it similarly to how we do for the distribution itself by separating JARs from other contents.

Reply:
> I totally missed the fact that these get installed after the node starts

Can you explain this more? We should not be touching plugins or modules after the node starts, and it would not work (ES only loads plugins at startup).

Reply:
That was poorly worded. More accurately, these get installed inside the `ElasticsearchNode.start()` method, but before we actually launch the ES process. So no, we don't actually muck with the installation after it's started. The key bit here is that this happens _after_ the testing task begins execution, so input snapshotting has already happened.

Reply:
This Pr is getting large, but we should probably improve this right after this is merged by having a configuration that inherits from testRuntime track all the plugins and modules that we need to install in the test cluster. That would make this tracking more straight forward as well

Reply:
> That would make this tracking more straight forward as well

I'm not convinced it would. A `Configuration` is nothing more than a `FileCollection` in this case as these are all unmanaged file dependencies. Also they are `.zip` files so we still have to deal with unpacking them and dealing with their contents intelligently. A `Configuraiton` would also not support any notion of a remote plugin configured via a `URI` so we'd have to manage that as well.